pkg/m3admin/namespace.go

```diff
@@ -0,0 +1,115 @@
+	_defaultRententionOptionBufferFutureNanos                        = int64(600000000000)
+	_defaultRententionOptionBufferPastNanos                          = int64(600000000000)
+	_defaultRententionOptionBlockDataExpiry                          = true
+	_defaultRententionOptionBlockDataExpiryAfterNotAccessPeriodNanos = int64(300000000000)
+	_defaultIndexOptionenabled                                       = true
+	_defaultIndexOptionBlockSizeNanos                                = int64(7200000000000)
+)
+
+//NamespaceCreate will create a namespace
+func (m *m3admin) NamespaceCreate(cluster *myspec.M3DBCluster) error {
+	namespaceCreateRequest := admin.NamespaceAddRequest{
+		Name: cluster.GetObjectMeta().GetName(),
+		Options: &namespace.NamespaceOptions{
+			BootstrapEnabled:  _defaultBootstrapEnabled,
+			FlushEnabled:      _defaultFlushEnabled,
+			WritesToCommitLog: _defaultWritesToCommitLog,
+			CleanupEnabled:    _defaultCleanupEnabled,
+			RepairEnabled:     _defaultRepairEnabled,
+			RetentionOptions: &namespace.RetentionOptions{
+				RetentionPeriodNanos:                     _defaultRententionOptionRetentionPeriodNanos,
+				BlockSizeNanos:                           _defaultRententionOptionBlockSizeNanos,
+				BufferFutureNanos:                        _defaultRententionOptionBufferFutureNanos,
+				BufferPastNanos:                          _defaultRententionOptionBufferPastNanos,
+				BlockDataExpiry:                          _defaultRententionOptionBlockDataExpiry,
+				BlockDataExpiryAfterNotAccessPeriodNanos: _defaultRententionOptionBlockDataExpiryAfterNotAccessPeriodNanos,
+			},
+			SnapshotEnabled: _defaultSnapshotEnabled,
+			IndexOptions: &namespace.IndexOptions{
+				Enabled:        _defaultIndexOptionenabled,
+				BlockSizeNanos: _defaultIndexOptionBlockSizeNanos,
+			},
+		},
+	}
+	data, err := json.Marshal(namespaceCreateRequest)
+	if err != nil {
+		return err
+	}
+	// TODO(PS) - Remove hardcoded references
+	url := fmt.Sprintf("http://%s.%s:7201%s", _m3coordinatorServiceName, cluster.GetNamespace(), _namespaceURI)
+	request, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+	request.Header.Add("Content-Type", "application/json")
+	if _, err := m.doHttpRequest(request); err != nil {
+		return err
+	}
+	logrus.WithField("namespace", namespaceCreateRequest.GetName()).Info("successfully created namespace")
+	return nil
+}
+
+// NamespaceGet will get a namespace
+func (m *m3admin) NamespaceGet(cluster *myspec.M3DBCluster) (*admin.NamespaceGetResponse, error) {
+	// TODO(PS) - Remove hardcoded references
+	url := fmt.Sprintf("http://%s.%s:7201%s", _m3coordinatorServiceName, cluster.GetNamespace(), _namespaceURI)
+	request, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := m.doHttpRequest(request)
+	if err != nil {
+		return nil, err
+	}
+	// fix itz
+	data := &admin.NamespaceGetResponse{}
+	defer resp.Body.Close()
```

Comment:
For the underlying connection to be reused, all the data from the response has to be read. Could you change this defer to the following - 

```
defer func() {
  ioutil.ReadAll(resp.Body)
  resp.Body.Close()
}()
```

Reply:
Good call, thanks